src/OneDrive.Sdk.Authentication.Common/Caching/CredentialCache.cs

```diff
@@ -20,8 +20,9 @@ public class CredentialCache
     {
         internal readonly IDictionary<CredentialCacheKey, AccountSession> cacheDictionary =
             new ConcurrentDictionary<CredentialCacheKey, AccountSession>();
+        internal CredentialCacheKey MostRecentlyUsedKey { get; set; }
 
-        private const int CacheVersion = 1;
+        private const int CacheVersion = 2;
```

Comment:
With this version revision, it seems like we should create CredentialCacheV2 which does its the new storage/management around the existing CredentialCache object so if we add a CredentialCacheV3 we don't multiplex the number of conditional checks that need to be done, instead we just create a series of very secure nesting dolls